File: 4week/dajung/python/0103_모스부호.py
#프로그래머스 level 0
#https://school.programmers.co.kr/learn/courses/30/lessons/120838

#길이가 1이상 1000이하인 문자열 letter 
def solution(letter):
    morse = { 
        '.-':'a','-...':'b','-.-.':'c','-..':'d','.':'e','..-.':'f',
        '--.':'g','....':'h','..':'i','.---':'j','-.-':'k','.-..':'l',
        '--':'m','-.':'n','---':'o','.--.':'p','--.-':'q','.-.':'r',
        '...':'s','-':'t','..-':'u','...-':'v','.--':'w','-..-':'x',
        '-.--':'y','--..':'z'
    }  
    splited = letter.split(" ")
    decryption = ""
    for i in splited:
        decryption += morse[i]
    return decryption

File: 4week/jihwan/javscript/0103_모스부호.js
// 프로그래머스 Level 1 jihwan: 0103_모스부호
// https://school.programmers.co.kr/learn/courses/30/lessons/120838

function solution(letter) {
  var answer = "";
  let morse = {
    ".-": "a",
    "-...": "b",
    "-.-.": "c",
    "-..": "d",
    ".": "e",
    "..-.": "f",
    "--.": "g",
    "....": "h",
    "..": "i",
    ".---": "j",
    "-.-": "k",
    ".-..": "l",
    "--": "m",
    "-.": "n",
    "---": "o",
    ".--.": "p",
    "--.-": "q",
    ".-.": "r",
    "...": "s",
    "-": "t",
    "..-": "u",
    "...-": "v",
    ".--": "w",
    "-..-": "x",
    "-.--": "y",
    "--..": "z",
  };
  let arr = letter.split(" ");

  for (j in arr) {
    for (i in morse) {
      if (arr[j] == i) {
        answer += morse[i];
      }
    }
  }

  return answer;
}
